cranelift-frontend/src/ssa.rs

```diff
@@ -527,31 +565,39 @@ impl SSABuilder {
     fn finish_predecessors_lookup(
         &mut self,
         func: &mut Function,
-        temp_arg_val: Value,
-        temp_arg_var: Variable,
+        sentinel: Value,
+        var: Variable,
         dest_ebb: Ebb,
     ) {
         let mut pred_values: ZeroOneOrMore<Value> = ZeroOneOrMore::Zero;
 
-        // Iterate over the predecessors.
-        for _ in 0..self.predecessors(dest_ebb).len() {
-            // For each predecessor, we query what is the local SSA value corresponding
-            // to var and we put it as an argument of the branch instruction.
-            let pred_val = self.results.pop().unwrap();
+        let num_results = self.results.len();
+        let num_predecessors = self.predecessors(dest_ebb).len();
+
+        // Determine how many predecessors are yielding unique, non-temporary Values.
+        for i in 0..num_predecessors {
+            let pred_val = self.results[num_results - 1 - i];
```

Comment:
This might be a bit hard to read... `self.results.iter().rev()...`?

Reply:
Ah yeah, that would definitely be better.